pkg/cache/tas_flavor.go

```diff
@@ -108,12 +108,21 @@ func (c *TASFlavorCache) snapshotForNodes(log logr.Logger, nodes []corev1.Node,
 	snapshot := newTASFlavorSnapshot(log, c.Levels)
 	nodeToDomain := make(map[string]utiltas.TopologyDomainID)
 	for _, node := range nodes {
-		levelValues := utiltas.LevelValues(c.Levels, node.Labels)
-		capacity := resources.NewRequests(node.Status.Allocatable)
-		domainID := utiltas.DomainID(levelValues)
-		snapshot.levelValuesPerDomain[domainID] = levelValues
-		snapshot.addCapacity(domainID, capacity)
-		nodeToDomain[node.Name] = domainID
+		ready := false
+		for _, cond := range node.Status.Conditions {
+			// Only healthy and ready to accept pods nodes are considered for scheduling calculation
+			ready = (cond.Type == corev1.NodeReady && cond.Status == corev1.ConditionTrue)
+		}
+		if ready {
+			levelValues := utiltas.LevelValues(c.Levels, node.Labels)
+			capacity := resources.NewRequests(node.Status.Allocatable)
+			domainID := utiltas.DomainID(levelValues)
+			snapshot.levelValuesPerDomain[domainID] = levelValues
+			snapshot.addCapacity(domainID, capacity)
+			nodeToDomain[node.Name] = domainID
```

Comment:
It would be great to invert this condition with `continue` to don't touch current implementation.

Reply:
+1